backend/src/main/kotlin/org/loculus/backend/controller/SubmissionController.kt

```diff
@@ -259,6 +267,12 @@ class SubmissionController(
             headers.add(HttpHeaders.CONTENT_ENCODING, compression.compressionName)
         }
 
+        val totalRecords = submissionDatabaseService.countReleasedSubmissions(organism)
+        headers.add("x-total-records", totalRecords.toString())
+        // There's a possibility that the totalRecords change between the count and the actual query
+        // this is not too bad, if the client ends up with a few more records than expected
+        // We just need to make sure the etag used is from before the count
+        // Alternatively, we could read once to file while counting and then stream the file
```

Comment:
Could you please create an issue so that we can fix this at some point? :)

Reply:
And mention the issue in the comment.

Reply:
Sure good idea, in the future you can also immediately create the issue yourself